easybuild/easyblocks/m/msm.py

```diff
@@ -0,0 +1,108 @@
+##
+# Copyright 2009-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MSM, implemented as an easyblock
+@author: Davide Vanzo (Vanderbilt University)
+"""
+import os
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import copy_dir, change_dir, mkdir, remove_dir
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+
+
+class EB_MSM(MakeCp):
+    """Support for building and installing MSM."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize MSM specific variables and purge previous installations."""
+        super(EB_MSM, self).__init__(*args, **kwargs)
+        self.sources_root = os.path.join(self.builddir, 'MSM_HOCR-%s' % self.version)
+
+        # Ensure that nothing has been left over from previous installation attempts.
+        # This is necessary here since directories must be created before building
+        # and not removed before the installation step.
+        remove_dir(self.installdir)
```

Comment:
@vanzod This is troublesome under `--module-only`, should be moved to `configure_step`?